app/models/risk_condition.rb

```diff
@@ -11,44 +11,107 @@ class RiskCondition < ApplicationRecord
     newRiskCondition = RiskCondition.new(options)
     if not newRiskCondition.save
       raise "Fail to create new risk condition with type #{type} for course #{course_id}"
     end
+    return newRiskCondition
   end
 
   def self.get_current_for_course(course_name)
     conditions = []
     course_id = Course.find_by(name: course_name).id
-    conditions_for_course = RiskCondition.where(course_id: course_id)
-    for type in RiskCondition.condition_types do
-      condition = conditions_for_course.where(condition_type: type[1]).order("version DESC").first
-      if not condition.nil?
-        conditions << condition
+    max_version = RiskCondition.get_max_version(course_id)
+    
+    return [] if max_version == 0
+    
+    conditions_for_course = RiskCondition.where(course_id: course_id, version: max_version)
+    condition_types = conditions_for_course.map { |condition| condition.condition_type.to_sym }
+    if condition_types.any? { |type| type == :no_condition_selected }
+      return []
+    else
+      return conditions_for_course
+    end
+  end
+
+  def self.update_current_for_course(course_name, params)
+    # The code below assumes params[type] contains a hash of the parameters needed for the type
+    # where type is a string corresponding to the four types defined at the top of this file
+    course_id = Course.find_by(name: course_name).id
+    max_version = RiskCondition.get_max_version(course_id)
+    # Is params empty?
+    if params.length == 0 and max_version == 0
+      # puts "case 1: max_version = 0 (no previous conditons have been set) and instructor doesn't want any at this point"
+      return []
+    elsif max_version > 0
+      previous_conditions = RiskCondition.where(course_id: course_id, version: max_version)
+      previous_types = previous_conditions.map { |c| c.condition_type }
+      if params.length == 0
+        unless previous_types.any? { |t| t == "no_condition_selected" }
+          # puts "case 2: previous conditions set to something and instructor doesn't want any this time"
+          create_condition_for_course_with_type(course_id, 0, {}, max_version + 1)
+        # else
+        #   puts "case 3: previous conditions set to nothing selected and instructor doesn't want any this time either"
+        end
+        # indicator row for "currently no conditions selected" that user doesn't need to access
+        return []
+      else
+        conditions = []
+        no_change = true
+        if params.length == previous_types.length
+          previous_conditions.map do |c|
+            unless params[c.condition_type] == c.parameters
+              no_change = false
+              break
+            end
+          end
+        else
+          no_change = false
+        end
+
+        unless no_change
+          # puts "case 4: instructor changed conditions this time; previous conditions were either unset, or different from current parameters"
+          params.map do |k, v|
```

Comment:
this is somehow not throwing an error for malformed bodys.

`{"no_submi312ssions":{"no_submissions_threshold":4}}`

Reply:
See latest commit.